src/analysis/inner_product_space/gram_schmidt_ortho.lean

```diff
@@ -90,116 +93,103 @@ begin
-    have h₂ := gram_schmidt_def' 𝕜 f n.succ,
-    simp only [nat.succ_eq_add_one, h₁, orthogonal_projection_singleton, zero_add] at h₂,
-    have h₃ : f (n + 1) ∈ span 𝕜 (f '' Iic n),
-    { rw [h₂, ← span_gram_schmidt 𝕜 f n],
-      apply submodule.sum_mem _ _,
-      simp_intros a ha only [finset.mem_range],
-      apply submodule.smul_mem _ _ _,
-      refine subset_span _,
-      simp only [mem_image, mem_Iic],
-      exact ⟨a, by linarith, by refl⟩, },
-    change linear_independent 𝕜 (f ∘ (coe : fin (n + 2) → ℕ)) at h₀,
-    have h₄ : ((n + 1) : fin (n + 2)) ∉ (coe : fin (n + 2) → ℕ) ⁻¹' (Iic n),
-    { simp only [mem_preimage, mem_Iic, not_le],
-      norm_cast,
-      rw fin.coe_coe_of_lt;
-      linarith, },
-    apply linear_independent.not_mem_span_image h₀ h₄,
-    rw [image_comp, image_preimage_eq_inter_range],
-    simp only [function.comp_app, subtype.range_coe_subtype],
-    convert h₃,
-    { norm_cast,
-      refine fin.coe_coe_of_lt (by linarith), },
-    { simp only [inter_eq_left_iff_subset, Iic, set_of_subset_set_of],
-      exact (λ a ha, by linarith), }, },
+  by_contra h,
+  have h₁ : f n ∈ span 𝕜 (f '' Iio n),
+  { rw [← span_gram_schmidt 𝕜 f n, gram_schmidt_def' _ f, h, zero_add],
+    apply submodule.sum_mem _ _,
+    simp_intros a ha only [finset.mem_Ico],
+    simp only [set.mem_image, set.mem_Iio, orthogonal_projection_singleton],
+    apply submodule.smul_mem _ _ _,
+    rw finset.mem_Iio at ha,
+    refine subset_span ⟨a, ha, by refl⟩ },
+  have h₂ : (f ∘ (coe : set.Iic n → ι)) ⟨n, le_refl n⟩
+    ∈ span 𝕜 (f ∘ (coe : set.Iic n → ι) '' Iio ⟨n, le_refl n⟩),
+  { rw [image_comp],
+    convert h₁ using 3,
+    ext i,
+    simpa using @le_of_lt _ _ i n },
+  apply linear_independent.not_mem_span_image h₀ _ h₂,
+  simp only [set.mem_Iio, lt_self_iff_false, not_false_iff]
 end
 
-/-- If the input of `gram_schmidt` is linearly independent, then the output is non-zero. -/
-lemma gram_schmidt_ne_zero' (f : ℕ → E) (h₀ : linear_independent 𝕜 f) (n : ℕ) :
+/-- If the input vectors of `gram_schmidt` are linearly independent,
+then the output vectors are non-zero. -/
+lemma gram_schmidt_ne_zero [succ_order ι] [is_succ_archimedean ι]
+    (f : ι → E) (n : ι) (h₀ : linear_independent 𝕜 f) :
   gram_schmidt 𝕜 f n ≠ 0 :=
-gram_schmidt_ne_zero 𝕜 f n (linear_independent.comp h₀ _ (fin.coe_injective))
+gram_schmidt_ne_zero_coe _ _ _ (linear_independent.comp h₀ _ subtype.coe_injective)
 
 /-- the normalized `gram_schmidt`
 (i.e each vector in `gram_schmidt_normed` has unit length.) -/
-noncomputable def gram_schmidt_normed (f : ℕ → E) (n : ℕ) : E :=
+noncomputable def gram_schmidt_normed (f : ι → E) (n : ι) : E :=
 (∥gram_schmidt 𝕜 f n∥ : 𝕜)⁻¹ • (gram_schmidt 𝕜 f n)
 
-lemma gram_schmidt_normed_unit_length (f : ℕ → E) (n : ℕ)
-  (h₀ : linear_independent 𝕜 (f ∘ (coe : fin n.succ → ℕ))) :
-    ∥gram_schmidt_normed 𝕜 f n∥ = 1 :=
+lemma gram_schmidt_normed_unit_length [succ_order ι] [is_succ_archimedean ι]
```

Comment:
This seems to be exactly analogous to Yael's point about the weaking of `gram_schmidt_ne_zero`.

Could you make a parallel change to 1bf2c32ac9c4ecab6cc23f01c24fef53eeafb324 here?

Reply:
Oh, of course, I missed that part.